src/pages/profile/Profile.jsx

```diff
@@ -1,58 +1,95 @@
-        <div className="profile__header__body-images">
-          <div className="background-img">
-            <img src={resultImg} alt="background" loading="lazy" />
+    <>
+      {singleUser?.username && (
+        <div className="profile__main">
+          <div className="profile__header">
+            <h3>Profile</h3>
           </div>
-          <div className="avatar-img">
-            <img src={avatar} alt="avatar" loading="lazy" />
+          <div className="profile__header__body">
+            <div className="profile__header__body-images">
+              <div className="background-img">
+                <img
+                  src={singleUser?.coverPhoto}
+                  alt="background"
+                  loading="lazy"
+                />
+              </div>
+              <div className="avatar-img">
+                <img src={singleUser?.userPhoto} alt="avatar" loading="lazy" />
+              </div>
+            </div>
+            <div className="profile-edit">
+              <button
+                className="btn btn-edit-profile"
+                onClick={() => setIsEditModalOpen(true)}
+              >
+                Edit Profile
+              </button>
+            </div>
+            <div className="profile__header__body-user">
+              <h3 className="profile-name">
+                {singleUser?.firstName} {singleUser?.lastName}
+              </h3>
+              <p className="profile-username">@{singleUser?.username}</p>
+            </div>
+
+            <div className="profile__header__body-description">
+              <p>{singleUser.bio}</p>
+              <a
+                href={singleUser?.link}
+                target="_blank"
+                style={{ color: "blue" }}
+              >
+                {singleUser?.link}
+              </a>
+            </div>
+            <div className="profile__header__body-footer">
+              <p className="follow-count">
+                {singleUser?.following?.length} Following
+              </p>
+              <p className="follow-count">
+                {singleUser?.followers?.length} Followers
+              </p>
+            </div>
           </div>
-        </div>
-        <div className="profile-edit">
-          <button className="btn btn-edit-profile">Edit Profile</button>
-        </div>
-        <div className="profile__header__body-user">
-          <h3 className="profile-name">Vishal Kumar</h3>
-          <p className="profile-username">@Vishalk01234</p>
-        </div>
 
-        <div className="profile__header__body-description">
-          <p>
-            Aspiring Full Stack developer 👨‍💻 | learning and sharing | neogcamp |
-            sometimes Shayar ✍️
-          </p>
-        </div>
-        <div className="profile__header__body-footer">
-          <p className="follow-count">10 Following</p>
-          <p className="follow-count">22M Followers</p>
+          <div className="profile__content">
+            <Post
```

Comment:
You Can use a loop to display their <Post /> component.
`data=[{your values},{your values}]`
`data.map((it)=>{
return (
<Post item={it} />
)
})`

Reply:
Yes, in upcoming PR I'll update the changes.